### PR TITLE
added init configuration

### PR DESCRIPTION
**Pull request by seleregb**
This PR addresses issue #48 

I added an extra command line option to receive `--init`, and upon the use of the option, a copy of the `appsetting.json` file from the app's base directory is copied to the user's current directory.

### CODE REVIEW

File: FakeServer/Program.cs [hunk 1]
```diff
@@ -122,15 +123,31 @@ private static int Run(string[] args, Dictionary<string, string> initialData)
                     app.RemainingArguments.Add(optionsUrls.Value());
                 }
 
+                if (optionInit.HasValue()) {
```
Comment (ttu) [inline FakeServer/Program.cs]: Use same formatting as the rest of the code base. Move curly brackets to new line.

File: FakeServer/Program.cs [hunk 2]
```diff
@@ -122,15 +123,31 @@ private static int Run(string[] args, Dictionary<string, string> initialData)
...
+                    app.RemainingArguments.Add("--init");
```
Comment (ttu) [inline FakeServer/Program.cs]: RemainingArguments are passed to server startup. App should exit after init so these are no needed.

File: FakeServer/Program.cs [hunk 3]
```diff
@@ -122,15 +123,31 @@ private static int Run(string[] args, Dictionary<string, string> initialData)
...
+                    app.RemainingArguments.Add(optionInit.Value());
+                    var baseAppSettingsFile = Path.Combine(AppDomain.CurrentDomain.BaseDirectory, "appsettings.json");
+                    var newAppSettingFile = Path.Combine(initialData["currentPath"],"appsettings.json");
+                    if (!File.Exists(newAppSettingFile)) {
```
Comment (ttu) [inline FakeServer/Program.cs]: If file already exiists, write to console something like: Can't create ne appsettings.json. File already exists at current path.

File: FakeServer/Program.cs [hunk 4]
```diff
@@ -122,15 +123,31 @@ private static int Run(string[] args, Dictionary<string, string> initialData)
...
+                        File.Copy(baseAppSettingsFile, newAppSettingFile);
+                    }
+                    Console.WriteLine($"AppSettings file created at default location: {initialData["currentPath"]}");
+                }
```
Comment (ttu) [inline FakeServer/Program.cs]: App should exit after init, so return 0;

File: FakeServer/Program.cs [hunk 5]
```diff
@@ -80,6 +80,7 @@ private static int Run(string[] args, Dictionary<string, string> initialData)
             var optionServe = app.Option("-s|--serve <PATH>", "Static files (default wwwroot)",
                 CommandOptionType.SingleValue);
             var optionsUrls = app.Option("--urls <URLS>", "Server url (default http://localhost:57602)", CommandOptionType.SingleValue);
+            var optionInit = app.Option("--init", "Initializes a new appsettings.json file", CommandOptionType.NoValue);
```
Comment (ttu) [inline FakeServer/Program.cs]: Description could be: Initializes a new appsettings.json file to current folder

File: FakeServer/Program.cs [hunk 6]
```diff
@@ -122,15 +123,31 @@ private static int Run(string[] args, Dictionary<string, string> initialData)
...
             return app;
-        }
+        } 
 
         private static string GetAssemblyVersion()
         {
             return FileVersionInfo.GetVersionInfo(Assembly.GetExecutingAssembly().Location).ProductVersion;
         }
+
+        private static string GetExecutingAssemblyPath()
```
Comment (ttu) [inline FakeServer/Program.cs]: Method is not used, so this can be removed.

**Comment by seleregb** [comment]
Thanks for the review. I've updated the changes accordingly.

File: FakeServer/Program.cs [hunk 7]
```diff
@@ -122,11 +123,27 @@ private static int Run(string[] args, Dictionary<string, string> initialData)
                     app.RemainingArguments.Add(optionsUrls.Value());
                 }
 
+                if (optionInit.HasValue()) 
+                {
+                    // app.RemainingArguments.Add("--init");
```
Comment (ttu) [inline FakeServer/Program.cs]: Remove commented lines

File: FakeServer/Program.cs [hunk 8]
```diff
@@ -122,11 +123,27 @@ private static int Run(string[] args, Dictionary<string, string> initialData)
...
+                    var baseAppSettingsFile = Path.Combine(AppDomain.CurrentDomain.BaseDirectory, "appsettings.json");
+                    var newAppSettingFile = Path.Combine(initialData["currentPath"],"appsettings.json");
+                    if (!File.Exists(newAppSettingFile)) 
+                    {
+                        File.Copy(baseAppSettingsFile, newAppSettingFile);
+                        Console.WriteLine($"AppSettings file created in current folder: {initialData["currentPath"]}");
+                        return 0;
+                    }
+                    
+                    Console.WriteLine("AppSettings.json file already exists in current folder!");
```
Comment (ttu) [inline FakeServer/Program.cs]: If file already exists should also return 

```cs
if (!File.Exists(newAppSettingFile)) 
{
  File.Copy(baseAppSettingsFile, newAppSettingFile);
  Console.WriteLine($"AppSettings file created in current folder: {initialData["currentPath"]}");
}
else
{
  Console.WriteLine("AppSettings.json file already exists in current folder!");
}

return 0;
```

**Comment by seleregb** [comment]
I've resolved the changes

**Comment by ttu** [comment]
Thanks for the implementation!

**Pull request merged**